src/SDKs/Search/DataPlane/Search.Tests/Utilities/EncryptionFixture.cs

```diff
@@ -0,0 +1,200 @@
+        {
+            ResourceManagementClient client = context.GetServiceClient<ResourceManagementClient>();
+            Provider provider = client.Providers.Register("Microsoft.KeyVault");
+        }
+
+        private static Vault CreateKeyVault(MockContext context, string resourceGroupName, string tenantId, string[] servicePrincipalIdsWithReadAccess)
+        {
+            var accessPolicies = new List<AccessPolicyEntry>();
+
+            // we add an access policy for each service principal that requires read access to the key vault. 
+            foreach (string servicePrincipalId in servicePrincipalIdsWithReadAccess)
+            {
+                accessPolicies.Add(new AccessPolicyEntry
+                {
+                    TenantId = System.Guid.Parse(tenantId),
+                    ObjectId = servicePrincipalId,
+                    Permissions = new Permissions()
+                    {
+                        Keys = new List<string>()
+                    {
+                        KeyPermissions.Get,
+                        KeyPermissions.WrapKey,
+                        KeyPermissions.UnwrapKey
+                    }
+                    }
+                });
+            }
+
+            string servicePrincipalObjectId = TestEnvironmentFactory.GetTestEnvironment().ConnectionString.KeyValuePairs[ConnectionStringKeys.AADClientIdKey];
+            // we also add an access policy for the sdk test account itself with full management permission to create and delete keys
+            accessPolicies.Add(new AccessPolicyEntry
+            {
+                TenantId = System.Guid.Parse(tenantId),
+                ObjectId = servicePrincipalObjectId,
+                Permissions = new Permissions()
+                {
+                    Keys = new List<string>()
+                    {
+                        KeyPermissions.Create,
+                        KeyPermissions.Delete,
+                        KeyPermissions.Get,
+                        KeyPermissions.List
+                    }
+                }
+            });
+
+            KeyVaultManagementClient keyVaultMgmtClient = context.GetServiceClient<KeyVaultManagementClient>();
+            string keyVaultName = TestUtilities.GenerateName("keyvault");
+            return keyVaultMgmtClient.Vaults.CreateOrUpdate(resourceGroupName, keyVaultName, new Microsoft.Azure.Management.KeyVault.Models.VaultCreateOrUpdateParameters
+            {
+                Location = "northcentralus",
+                Properties = new Microsoft.Azure.Management.KeyVault.Models.VaultProperties
+                {
+                    TenantId = System.Guid.Parse(tenantId),
+                    AccessPolicies = accessPolicies,
+                    Sku = new Microsoft.Azure.Management.KeyVault.Models.Sku(Microsoft.Azure.Management.KeyVault.Models.SkuName.Standard)
+                }
+            });
+        }
+
+        private static KeyBundle CreateKey(MockContext context, Vault keyVault)
```

Comment:
I guess that's why I created the EncryptionFixture class, to abstract away all the key-vault related code so it wouldn't polluate the actual test (or the search service fixture) with details unrelated to the usage of the azure search SDK. I could rename the fixture "KeyVaultFixture" if it helps with understanding the context better. I had decided on "EncryptionFixture" so it would be clear to what Azure Search feature this fixture was related, but I don't mind changing it